Classes/Projection/ProjectionManager.php

```diff
@@ -0,0 +1,214 @@
+        }
+        $this->projectionCache->set($cacheId, $projectionState);
+        \TYPO3\Flow\var_dump($projectionState);
+
+    }
+
+    /**
+     * Takes a short projection identifier and returns the "full" identifier if valid
+     *
+     * @param string $projectionIdentifier in the form "<package.key>:<projection>", "<key>:<projection>" or "<projection">"
+     * @return string
+     * @throws \InvalidArgumentException if no matching projector could be found
+     */
+    private function normalizeProjectionIdentifier($projectionIdentifier)
+    {
+        $matchingIdentifiers = [];
+        foreach (array_keys($this->projections) as $fullProjectionIdentifier) {
+            if ($this->projectionIdentifiersMatch($projectionIdentifier, $fullProjectionIdentifier)) {
+                $matchingIdentifiers[] = $fullProjectionIdentifier;;
+            }
+        }
+        if ($matchingIdentifiers === []) {
+            throw new \InvalidArgumentException(sprintf('No projection could be found that matches the projection identifier "%s"', $projectionIdentifier), 1476368605);
+        }
+        if (count($matchingIdentifiers) !== 1) {
+            throw new \InvalidArgumentException(sprintf('More than one projection matches the projection identifier "%s":%s%s', $projectionIdentifier, chr(10), implode(', ', $matchingIdentifiers)), 1476368615);
+        }
+        return $matchingIdentifiers[0];
+    }
+
+    /**
+     * Determines whether the given $shortIdentifier in the form "<package.key>:<projection>", "<key>:<projection>" or "<projection">"
+     * matches the $fullIdentifier in the form "<package.key>:<projection>"
+     *
+     * @param string $shortIdentifier
+     * @param string $fullIdentifier The full projection identifier
+     * @return bool
+     * @throws \InvalidArgumentException if the given $shortIdentifier is not in the valid form
+     * @see normalizeProjectionIdentifier()
+     */
+    private function projectionIdentifiersMatch(string $shortIdentifier, string $fullIdentifier): bool
+    {
+        $shortIdentifier = strtolower($shortIdentifier);
+        if ($shortIdentifier === $fullIdentifier) {
+            return true;
+        }
+        $shortIdentifierParts = explode(':', $shortIdentifier);
+        $fullIdentifierParts = explode(':', $fullIdentifier);
+
+        $shortIdentifierPartsCount = count($shortIdentifierParts);
+        if ($shortIdentifierPartsCount === 1) {
+            return $shortIdentifier === $fullIdentifierParts[1];
+        }
+        if ($shortIdentifierPartsCount !== 2) {
+            throw new \InvalidArgumentException(sprintf('Invalid projection identifier "%s", identifiers must have the format "<projection>" or "<package-key>:<projection>".', $shortIdentifier), 1476367741);
+        }
+        return
+            $shortIdentifierParts[1] === $fullIdentifierParts[1]
+            && substr($fullIdentifierParts[0], -(strlen($shortIdentifierParts[0]) + 1)) === '.' . $shortIdentifierParts[0];
```

Comment:
> Do we really need to support all those different half-explicit spellings

I'd say: Yes, absolutely ;)
It would be a pain if one had to write
```
flow typo3.kickstart:kickstart:commandcontroller...
```
or 
```
typo3.flow:doctrine:migrationgenerate
```
for example. And the same with
```
flow projection:replay some.package.key:foo
```
IMO.
We could obviously move that logic somewhere else, but I don't think that it does any harm. It's just private methods after all